cmd/fleet/cron.go

```diff
@@ -1074,6 +1074,24 @@ func newAppleMDMDEPProfileAssigner(
 				return nil
 			}
 
+			// As part of the DB migration of the single ABM token to the multi-ABM
+			// token world (where the token was migrated from mdm_config_assets to
+			// abm_tokens), we need to complete migration of the existing token as
+			// during the DB migration we didn't have the organization name, apple id
+			// and renewal date.
+			incompleteToken, err := ds.GetABMTokenByOrgName(ctx, "")
```

Comment:
There's a little risk here that while the token has no org name, another ABM token gets added with an org name that would conflict with this one (and we'll only know once we get the org name for that migrated token, at which point the SaveABMToken would fail due to the unique constraint).

Thoughts on how to handle that? I think that's a pretty unlikely edge case, the chances of it happening AFAICT are:

* if Apple's API is down for a while when Fleet restarts after the DB migration (so the migrated token would not be updated until some minutes later when it comes back up, giving time to a user to add a new token but then adding a token also requires that Apple's API is up, so the window is small)
* if the terms have expired for that token, which is similar to if Apple's API is down but only for that token. But then again, for a user to be able to add a new conflicting token (with the same org name), it would probably not be possible without accepting the new terms first (and then it's once again a race with the cron job, a small window).
* if the user is super fast and goes to the UI to add a new conflicting token before that cron runs (at Fleet startup, it should run in the first 10 seconds!)

Is this big enough a risk to worry about it? Worse case we'd end up with two tokens, one with an Org Name, one without org name/renew date, so it would show up in Fleet as expired (and the user could delete it from the UI, or delete the newer one they added). I _think_ the rest would still work (depends if in the cron job we use only tokens that have a non-expired renew date).

If we feel this needs to be handled, the easiest way would be to fail adding new tokens until there is no existing one without an org name.

Reply:
in my opinion, it's not worth handling this case, this cron runs every 30 seconds anyways, so any scenario that involves the API down or the user accepting the new ToS it's extremely unlikely